### PR TITLE
Small tweak that saves two bytes

### PR DESCRIPTION
**Pull request by TerjeWiigMathisen**
Saw your Smallest Windows asm program video and took a look at the source code, found one obvious improvement

Regards, Terje Mathisen

### CODE REVIEW

**Comment by rbergen** [comment]
That didn't take long. :)

Thanks for submitting this, but could you please:

- Convert the comments behind the original lines into the comment behind your line
- Remove the original, now commented out, lines
- Use the same spacing style in your code as the rest of the code does

Thank you.

**Comment by TerjeWiigMathisen** [comment]
Sure, no problem!

**Comment by rbergen** [comment]
Thanks. Concerning the spacing, I meant the whitespacing between the various parts of the operation argument. I know it may not be a very common whitespacing style for assembly, but it can help with the readability for people less familiar with assembly's dense style. I can make the change myself, but that will constitute an edit in your branch.

**Comment by rbergen** [comment]
I see you're still working on the PR. That's great, but then I will have to ask you to apply the whitespacing coding style used in the file, yourself.

**Comment by TerjeWiigMathisen** [comment]
I inspected your code in notepad++, and it looks like only spaces, no tabs.
Is that wrong, or did I mess up while making my fork?

Sorry!

**Comment by rbergen** [comment]
No, what I mean is that the arguments in `lea ebx,[edi-sp_egg]` should be spaced as `ebx, [edi - sp_egg]`, so with spaces between the various parts of the . Same thing for the other change you made. Please just take a look at the other operation arguments in the file you're editing, and apply that same style.

**Comment by TerjeWiigMathisen** [comment]
Ok, thank you!

Having written literally Megabytes of asm I never even noticed that. Will
fix!

**Comment by rbergen** [comment]
I do understand that. As I said earlier, I acknowledge that the spacing applied here is not necessarily the most common.

File: Lasse/LittleWindows.asm [hunk 1]
```diff
@@ -81,8 +84,7 @@ option casemap:none                             ; Preserve the case of system id
 
 ; Setting up our own stack frame
 start proc
-        mov ebp, esp
-        add esp, sp_inv_main                    ; this gives 240 bytes, more than enough for our variables
+        enter -sp_inv_main, 0                   ; this gives 240 bytes, more than enough for our variables
```
Comment (rbergen) [inline Lasse/LittleWindows.asm:87]: We're now inverting the inverted stack frame size. I think it's then better to change the define for `sp_inv_main` in LittleWindows.inc to its positive counterpart (i.e. 240), rename it to `sp_main` and change this line of code accordingly. Same thing for the WndProc stack frame as well.

File: Lasse/LittleWindows.asm [hunk 2]
```diff
@@ -166,12 +168,10 @@ find_egg:
...
         cmp dword ptr ds:[edi], eax             ; check for "DAVE"
         jne find_egg                            ; loop if not found
-        add edi, 4                              ; move to next 4 bytes
-        cmp dword ptr ds:[edi], eax             ; check for "DAVE" again
+        cmp dword ptr ds:[edi + 4], eax         ; check for "DAVE" again
         jne find_egg                            ; loop if not found
 matched:
-        mov ebx, edi                            ; put the place we found in ebx...
-        sub ebx, sp_egg + 4                     ; ...and adjust it
+        lea ebx,[edi - sp_egg]                  ; return the adjusted place we found
```
Comment (rbergen) [inline Lasse/LittleWindows.asm:174]: Please add a space after the comma.

File: Lasse/LittleWindows.asm [hunk 3]
```diff
@@ -415,9 +415,7 @@ WinMainRet:
 
 WndProc:
         call egghunter                          ; ebp is incorrect at this point. We call our egghunter function to reposition it, and place it into ebx
-        push ebp                                ; we adheare to rules of stdcall
-        mov ebp, esp                            ; we setup a new stack frame
-        add esp, sp_inv_WndProc                 ; we need 84 bytes of space
+        enter -sp_inv_WndProc, 0                ; use stdcall, setup a new stack frame of 84 bytes
```
Comment (rbergen) [inline Lasse/LittleWindows.asm:418]: See first comment.

File: Lasse/LittleWindows.asm (hunk 1 above)
Comment (TerjeWiigMathisen) [reply Lasse/LittleWindows.asm]: That's a good idea, done.

File: Lasse/LittleWindows.asm (hunk 2 above)
Comment (TerjeWiigMathisen) [reply Lasse/LittleWindows.asm]: Did so, I think/hope this was the last formatting error

**Comment by rbergen** [comment]
Thanks! I can confirm that this takes 16 bytes off the executable without the use of Crinkler, and 8 bytes when Crinkler is used. The executable that is generated also still works on my machine.

Please let me know if you've concluded your work on this. If so, I'm happy to merge.

**Comment by lassehauballe** [comment]
Just tested it, and it runs perfectly. 

Must admit that I am shaking my head over the egghunter optimization :D 

Nice finding!

**Comment by TerjeWiigMathisen** [comment]
I am done with this batch, looking into more substantial changes!
Will create a new branch & pr for those if they work out.

Regards,
Terje

**Comment by rbergen** [comment]
If you do find other ways to save space, we look forward to your next PR. Until then, thank you for this contribution!

**Pull request merged**